homeassistant/components/binary_sensor/ihc.py

```diff
@@ -0,0 +1,183 @@
+"""IHC binary sensor platform."""
+# pylint: disable=too-many-instance-attributes
+import logging
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.binary_sensor import (
+    BinarySensorDevice, PLATFORM_SCHEMA, DEVICE_CLASSES_SCHEMA)
+from homeassistant.const import (STATE_UNKNOWN, CONF_NAME, CONF_TYPE,
+                                 CONF_ID, CONF_BINARY_SENSORS)
+
+from homeassistant.components.ihc.const import CONF_AUTOSETUP, CONF_INVERTING
+from homeassistant.components.ihc import get_ihc_platform
+from homeassistant.components.ihc.ihcdevice import IHCDevice
+
+DEPENDENCIES = ['ihc']
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_AUTOSETUP, default='False'): cv.boolean,
+    vol.Optional(CONF_BINARY_SENSORS):
+        [{
+            vol.Required(CONF_ID): cv.positive_int,
+            vol.Optional(CONF_NAME): cv.string,
+            vol.Optional(CONF_TYPE): DEVICE_CLASSES_SCHEMA,
+            vol.Optional(CONF_INVERTING): cv.boolean,
+        }]
+})
+
+
+PRODUCTAUTOSETUP = [
+    # Magnet contact
+    {'xpath': './/product_dataline[@product_identifier="_0x2109"]',
+     'node': 'dataline_input',
+     'type': 'opening',
+     'inverting': True},
+    # Pir sensors
+    {'xpath': './/product_dataline[@product_identifier="_0x210e"]',
+     'node': 'dataline_input',
+     'type': 'motion',
+     'inverting': False},
+    # Pir sensors twilight sensor
+    {'xpath': './/product_dataline[@product_identifier="_0x0"]',
+     'node': 'dataline_input',
+     'type': 'motion',
+     'inverting': False},
+    # Pir sensors alarm
+    {'xpath': './/product_dataline[@product_identifier="_0x210f"]',
+     'node': 'dataline_input',
+     'type': 'motion',
+     'inverting': False},
+    # Smoke detector
+    {'xpath': './/product_dataline[@product_identifier="_0x210a"]',
+     'node': 'dataline_input',
+     'type': 'smoke',
+     'inverting': False},
+    # leak detector
+    {'xpath': './/product_dataline[@product_identifier="_0x210c"]',
+     'node': 'dataline_input',
+     'type': 'moisture',
+     'inverting': False},
+    # light detector
+    {'xpath': './/product_dataline[@product_identifier="_0x2110"]',
+     'node': 'dataline_input',
+     'type': 'light',
+     'inverting': False},
+]
+
+_LOGGER = logging.getLogger(__name__)
+_IHCBINARYSENSORS = {}
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the IHC binary setsor platform."""
+    ihcplatform = get_ihc_platform(hass)
```

Comment:
Changed to
```python
    ihc = hass.data[IHC_DATA]
```
the get_ihc_platform is not needed (as you also say further down), this was because I did not have the dependency setup and then binary sensor could be loaded before the ihc component